src/main/java/ca/on/oicr/gsi/dimsum/util/reporting/reports/CaseTatReport.java

```diff
@@ -0,0 +1,206 @@
+                      .ofNullable(findLatestCompletionDate(x.getTest().getFullDepthSequencings()))
+                      .map(date -> date.format(DateTimeFormatter.ISO_LOCAL_DATE)).orElse("")),
+              Column.forInteger("Full-Depth Days",
+                  x -> x.getTest().getFullDepthSequencingDaysSpent()),
+              Column.forString("Analysis Review Completed", x -> {
+                List<LocalDate> dates = x.getCase().getDeliverables().stream()
+                    .map(CaseDeliverable::getAnalysisReviewQcDate).collect(Collectors.toList());
+                if (dates.contains(null)) {
+                  return null;
+                }
+                return dates.stream().max(LocalDate::compareTo)
+                    .map(date -> date.format(DateTimeFormatter.ISO_LOCAL_DATE)).orElse(null);
+              }),
+              Column.forInteger("Analysis Review Days",
+                  x -> x.getCase().getAnalysisReviewDaysSpent()),
+              Column.forString("Release Approval Completed", x -> {
+                List<LocalDate> dates = x.getCase().getDeliverables().stream()
+                    .map(CaseDeliverable::getReleaseApprovalQcDate).collect(Collectors.toList());
+                if (dates.contains(null)) {
+                  return null;
+                }
+                return dates.stream().max(LocalDate::compareTo)
+                    .map(date -> date.format(DateTimeFormatter.ISO_LOCAL_DATE)).orElse(null);
+              }),
+              Column.forInteger("Release Approval Days",
+                  x -> x.getCase().getReleaseApprovalDaysSpent()),
+              Column.forString("Completion Date", x -> getCompletionDate(x.getCase())),
+              Column.forInteger("Total Days", x -> x.getCase().getCaseDaysSpent()))) {
+
+        @Override
+        public List<RowData> getData(CaseService caseService, Map<String, String> parameters) {
+          List<CaseFilter> filters = convertParametersToFilters(parameters);
+          return caseService.getCaseStream(filters)
+              .flatMap(kase -> kase.getTests().stream().map(test -> new RowData(kase, test)))
+              .collect(Collectors.toList());
+        }
+      };
+
+  public static final CaseTatReport INSTANCE = new CaseTatReport();
+
+  private CaseTatReport() {
+    super("Case TAT Report", caseSection);
+  }
+
+  private static String getSortedProjectNameOrPipeline(Case kase,
+      Function<Project, String> function) {
+    return kase.getProjects().stream()
+        .map(function)
+        .sorted()
+        .collect(Collectors.joining(", "));
+  }
+
+  private static List<CaseFilter> convertParametersToFilters(Map<String, String> parameters) {
+    List<CaseFilter> filters = new ArrayList<>();
+    parameters.forEach((key, value) -> {
+      CaseFilterKey filterKey = CaseFilterKey.valueOf(key.toUpperCase());
+      CaseFilter filter = new CaseFilter(filterKey, value);
+      filters.add(filter);
+    });
+    return filters;
+  }
```

Comment:
* Should catch `IllegalArgumentException` from `CaseFilterKey#valueOf` and rethrow as `BadRequestException` (IllegalArgument would result in a server (back-end) error, whereas BadRequest is a front-end error)
* The front end may send multiple values separated by commas. This is not handled here